index.js

```diff
@@ -106,7 +106,7 @@ module.exports._flush = function(lastTime, code) {
 
   return RSVP.Promise.resolve(lastTime).
     then(function() {
-      return RSVP.Promise.all(work.map(function(handler) {
+      return RSVP.allSettled(work.map(function(handler) {
```

Comment:
this isn't correct, as if any of them reject. We will now not correctly reject.

I am OK with exploring using `allSettled` so we wait for all the settle, but if any of them failed we should still reject the final promise. This gets tricky when it becomes a rollup of many rejections..

This is nuanced area, we should think through this carefully. But as currently written, legit failures will be hidden.